pkg/autoscaler/metrics/stats_scraper.go

```diff
@@ -194,29 +194,27 @@ func urlFromTarget(t, ns string) string {
 // Scrape calls the destination service then sends it
 // to the given stats channel.
 func (s *serviceScraper) Scrape(window time.Duration) (Stat, error) {
-	readyPodsCount, err := s.podAccessor.ReadyCount()
-	if err != nil {
-		return emptyStat, ErrFailedGetEndpoints
-	}
-
-	if readyPodsCount == 0 {
-		return emptyStat, nil
-	}
-
 	startTime := time.Now()
 	defer func() {
 		scrapeTime := time.Since(startTime)
 		pkgmetrics.RecordBatch(s.statsCtx, scrapeTimeM.M(float64(scrapeTime.Milliseconds())))
```

Comment:
Should we avoid reporting metrics if we actually do not scrape?

Reply:
no pods — means no concurrency, so empty stat makes sense, yes.

Reply:
But actually we are recording here for no pods case.

Reply:
you mean scraping metrics?
I see, you mean scraping metrics — not sure matters much. We spent 0 time scraping, so it should be ok...

Reply:
It's a distribution. So data points with 0 time will make P95, P99 etc lower than actual values.

Reply:
Alright, let me see what I can do.